components/bl-dashlet-component/src/local-settings.js

```diff
@@ -0,0 +1,33 @@
+export class LocalSettings {
+  constructor(instanceId) {
+    this.instanceId = instanceId;
+  }
+
+  get isOpen() {
+    const isOpen = localStorage.getItem(this.instanceId + ' isOpen');
+
+    return isOpen === null || isOpen === 'true';
+  }
+
+  set isOpen(value) {
+    localStorage.setItem(this.instanceId + ' isOpen', String(value));
```

Comment:
It will flood the LS with a bunch of keys.

I would recommend you collect all the keys under a single key.

```
class Storage  {
 constructor(instanceId){
  this.name = `UIComponent_Dashlet_${instanceId}`
 }

 getStore(){
  let result

  try{
    result = localStorage.getItem(this.name);
} catch(e){
 result = {}
}

if(!result || typeof !== 'object' || Array.isArray(result)){
  result = {}
}

return result
 }

 setValue(key, value){
  const store = this.getStore()

  localStorage.setItem(this.name, {...store, [key]:value });
}

 getValue(key){
  const store = this.getStore()

  return store[key]
}

get isOpen() {
    return this.getValue('isOpen')
  }

  set isOpen(value) {
   this.setValue('isOpen', value)
}
...
}
```